packages/core/src/App/Containers/Modals/app-modals.jsx

```diff
@@ -48,12 +51,25 @@ const AppModals = ({
     is_close_uk_account_modal_visible,
     is_eu,
     is_logged_in,
+    should_show_cooldown_modal,
+    should_show_assessment_complete_modal,
+    is_trading_assessment_for_new_user_enabled,
+    fetchFinancialAssessment,
+    setCFDScore,
+    cfd_score,
+    active_account_landing_company,
     is_deriv_account_needed_modal_visible,
     is_warning_scam_message_modal_visible,
 }) => {
     const url_params = new URLSearchParams(useLocation().search);
     const url_action_param = url_params.get('action');
 
+    React.useEffect(() => {
+        if (is_logged_in) {
+            fetchFinancialAssessment().then(response => setCFDScore(response?.cfd_score ?? 0));
+        }
+    }, [is_logged_in]);
```

Comment:
do we really need to use here api calls instead of just passing the value? maybe it is better to do it in parent component? this func is expected to be called anytime being logged it for any users?

Reply:
The API call to get CFD score is better here coz, there is a requirement that when existing user logs in, if his CFD score is 0 and is an MF user, we need to ask the user to fill trading assessment. since the modal is triggered from App-modal, the API call is made here